AK/BigInts.h

```diff
@@ -0,0 +1,546 @@
+        return m_low;
+    }
+
+    Span<u8> bytes()
+    {
+        return Span<u8>(reinterpret_cast<u8*>(this), sizeof(this));
+    }
+    Span<const u8> bytes() const
+    {
+        return Span<const u8>(reinterpret_cast<const u8*>(this), sizeof(this));
+    }
+
+    // Comparison Operations
+    // FIXME: safe bool casting !!"" looks weird and "" == 1u is long
+    constexpr bool operator!()
+    {
+        return !m_low && !m_high;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator==(const T& other) const
+    {
+        return !m_high && m_low == other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator!=(const T& other) const
+    {
+        return m_high || m_low != other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator>(const T& other) const
+    {
+        return m_high || m_low > other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator<(const T& other) const
+    {
+        return !m_high && m_low < other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator>=(const T& other) const
+    {
+        return *this == other || *this > other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr bool operator<=(const T& other) const
+    {
+        return *this == other || *this < other;
+    }
+
+    constexpr bool operator==(const R& other) const
+    {
+        return m_low == other.low() && m_high == other.high();
+    }
+    constexpr bool operator!=(const R& other) const
+    {
+        return m_low != other.low() || m_high != other.high();
+    }
+    constexpr bool operator>(const R& other) const
+    {
+        return m_high > other.high()
+            || (m_high == other.high() && m_low > other.low());
+    }
+    constexpr bool operator<(const R& other) const
+    {
+        return m_high < other.high()
+            || (m_high == other.high() && m_low < other.low());
+    }
+    constexpr bool operator>=(const R& other) const
+    {
+        return *this == other || *this > other;
+    }
+    constexpr bool operator<=(const R& other) const
+    {
+        return *this == other || *this < other;
+    }
+
+    // Bitwise operations
+    constexpr R&& operator~()
+    {
+        return { ~m_low, ~m_high };
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr U&& operator&(const U& other) const
+    {
+        return static_cast<U>(m_low) & other;
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr R&& operator|(const U& other) const
+    {
+        return { m_low | other, m_high };
+    }
+    template<Unsigned U>
+    requires(sizeof(T) >= sizeof(U)) constexpr R&& operator^(const U& other) const
+    {
+        return { m_low ^ other, m_high };
+    }
+    template<Unsigned U>
+    constexpr R&& operator<<(const U& other) const
```

Comment:
`other` seems like a misnomer to me for the shift operators, `shift` would be easier to read.